conversations/impl/src/main/java/org/sakaiproject/conversations/impl/ConversationsServiceImpl.java

```diff
@@ -480,7 +480,21 @@ public TopicTransferBean saveTopic(final TopicTransferBean topicBean, boolean se
             if (sendMessage) {
                 try {
                     Site site = siteService.getSite(decoratedBean.siteId);
-                    Set<User> users = new HashSet<>(userDirectoryService.getUsers(site.getUsers()));
+
+                    Set<User> users = null;
+                    switch (topic.getVisibility()) {
+                        case SITE:
+                            users = new HashSet<>(userDirectoryService.getUsers(site.getUsers()));
+                            break;
+                        case GROUP:
+                            Set<String> userIds = new HashSet<>(authzGroupService.getAuthzUsersInGroups(topic.getGroups()));
+                            users = new HashSet<>(userDirectoryService.getUsers(userIds));
+                            break;
+                        case INSTRUCTORS:
+                            userIds = site.getUsersIsAllowed(Permissions.ROLETYPE_INSTRUCTOR.label);
+                            users = new HashSet<>(userDirectoryService.getUsers(userIds));
```

Comment:
It's good practise to have an explicit break in each case, unless you really want multiple cases running. In this case, if someone comes along and adds something to the default case without realizing they need to add a break in the above case, both would run (unintentionally).